app/controllers/EventSelectionController.scala

```diff
@@ -71,9 +84,25 @@ class EventSelectionController @Inject()(val controllerComponents: MessagesContr
                     reportVersion = answers.eventDataIdentifier(eventType).version))
                 Redirect(EventSelectionPage.navigate(waypoints, answers, answers).route)
               }
-            case _ => Future.successful(Ok(view(form, waypoints)))
+            case _ =>
+              eventReportingConnector.getFeatureToggle(ltaAbolitionShowHideToggle).flatMap { toggleData =>
+                val displayEventList: Seq[RadioItem] = getFilteredOptions(toggleData.isEnabled, request.userAnswers)
+                Future.successful(Ok(view(form, displayEventList, waypoints)))
+              }
           }
         }
       )
   }
+
+  private def getFilteredOptions(isEnabled: Boolean, ua: UserAnswers)(implicit messages: Messages): Seq[RadioItem] = {
```

Comment:
Does this def cover all the scenarios?

- Toggle on, >=2024
- Toggle on, < 2024
- Toggle off, >= 2024
- Toggle off, < 2024

Reply:
yes, working as expected